test/trigger/test_trigger.js

```diff
@@ -390,6 +405,136 @@ let testBadZoneId = function(file, contents) {
     errorFunc(`${file}: use zoneId instead of zoneRegex`);
 };
 
+// responses_test.js will handle testing any response with builtInResponseStr.
+// triggers using `response:` otherwise cannot be tested, because we cannot
+// safely call the response function.
+const testOutputStrings = (file, contents) => {
+  const json = eval(contents);
+  const triggerSet = json[0];
+  for (const set of [triggerSet.triggers, triggerSet.timelineTriggers]) {
+    if (!set)
+      continue;
+    for (const trigger of set) {
+      let outputStrings = {};
+      let response = {};
+      if (trigger.response) {
+        // Triggers using responses should include the outputStrings in the
+        // response func itself, via `output.responseOutputStrings = {};`
+        if (trigger.outputStrings) {
+          errorFunc(`${file}: found both 'response' and 'outputStrings in '${trigger.id}'.`);
+          continue;
+        }
+        if (typeof trigger.response !== 'function')
+          continue;
+        const funcStr = trigger.response.toString();
+        if (!funcStr.includes(builtInResponseStr))
+          continue;
+        const output = { responseOutputStrings: undefined };
+        // Call the function to get the outputStrings.
+        response = trigger.response({}, {}, output);
+        outputStrings = output.responseOutputStrings;
+
+        if (typeof outputStrings !== 'object') {
+          errorFunc(`${file}: '${trigger.id}' built-in response did not set outputStrings.`);
+          continue;
+        }
+      } else {
+        if (trigger.outputStrings && typeof outputStrings !== 'object') {
+          errorFunc(`${file}: '${trigger.id}' outputStrings must be an object.`);
+          continue;
+        }
+        outputStrings = trigger.outputStrings;
+      }
+
+      // TODO: should we prevent `output['phrase with spaces']()` style constructions?
```

Comment:
Is there a specific scenario you have in mind where this could be an issue, or wou8ld this just be to keep things consistent with the rest of the repository? I have no strong feelings on it, just curious.

Reply:
No, I think it was kind of a question.  I think the regex will match anything that's not a closing curly brace, but it was more a question of consistency.